stdlib/Random/src/Random.jl

```diff
@@ -434,4 +434,10 @@ true
 """
 seed!(rng::AbstractRNG, ::Nothing) = seed!(rng)
 
+# Randomize quicksort pivot selection. This code is here because of bootstrapping:
+# we need to sort things before we load this standard library.
+# TODO move this into Sort.jl
```

Comment:
now that the compiler's sort is separated, is this still necessary?

Reply:
Yes. If someone filters out the Random stdlib, I'd like to use hash-based pivot selection rather than error trying to use `rand`.

Reply:
If sorting were split into a small portion in base that works alone and a larger stdlib that pirates base's sorting to make it run a bit faster and depends on Random, then this code would be able to move to the sorting stdlib, so the TODO is still appropriate.

Reply:
And if you mean can we remove that redefinition entirely, yes we can. But it would come at a performance penalty (likely negligible) and at the cost of allowing pathological inputs to be intentionally created.